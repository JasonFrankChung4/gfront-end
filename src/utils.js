/*global ASSET_PATH */

import get from 'lodash.get'
import intersection from 'lodash.intersection'
import { run, startsWith } from '@code.gov/cautious'

export const falses = [undefined, null, 'null', 'None', 'Null', 'NULL', '', 'False', 'false']

export function isFalse(input) {
  return falses.includes(input)
}

export function isSet(input) {
  return input && typeof input === 'object' && input.has && input.add
}

export function adjustAssetPath(thing) {
  const pattern = /.?\/?assets\//
  const newAssetPath = ASSET_PATH + 'assets/'
  if (startsWith(thing, './assets') || startsWith(thing, '/assets/') || startsWith(thing, 'assets/')) {
    return thing.replace(pattern, newAssetPath)
  } else if (typeof value === 'object') {
    for (let key in thing) {
      const subvalue = thing[key]
      if (typeof subvalue === 'string') {
        thing[key] = adjustAssetPath(subvalue)
      }
    }
    return thing
  } else {
    return thing
  }
}

export function getConfigValue(siteConfig, path) {
  if (siteConfig) {
    let value = get(siteConfig, path)
    if (typeof value === 'string') {
      try {
        value = adjustAssetPath(value)
      } catch (error) {
        console.error("couldn't check if the following started with assets", [value])
      }
    } else if (Array.isArray(value)) {
      value = value.map(item => {
        if (typeof item === 'object') {
          for (let subkey in item) {
            let subvalue = item[subkey]
            if (typeof subvalue === 'string') {
              item[subkey] = adjustAssetPath(subvalue)
            }
          }
        }
        return item
      });
    }
    if (!value) {
      console.warn(`We weren't able to find the value for ${path} in your code-gov-config.json file.
      You can consult examples under the folder found at config/site/examples
      or consult our developer documentation here: https://developers.code.gov/configure.html`)
    }
    return value
  }
  else {
    return null
  }
}

export function normalize(input) {
  return trim(lower(input))
}

export function onEachItem(input, func) {
  if (Array.isArray(input)) {
    return input.map(item => run(item, func))
  } else if (isSet(input)) {
    return Set(Array.from(input).map(item => run(item, func)))
  } else {
    return run(input, func)
  }
}

export function lower(input) {
  return onEachItem(input, 'toLowerCase')
}

export function upper(input) {
  return onEachItem(input, 'toUpperCase')
}

export function trim(input) {
  return onEachItem(input, 'trim')
}

/* runs when each page component is loaded */
export function refreshView() {
  window.scrollTo(0, 0)
  document.activeElement.blur()
}


export function scrollToTopOfResults() {
  const headerHeight = document.querySelector("header.main").clientHeight
  const bannerHeight = document.querySelector(".banner").clientHeight
  const navHeight = document.querySelector("header.main nav").clientHeight
  const scrollDepth = headerHeight + bannerHeight - navHeight
  window.scrollTo(0, scrollDepth)
}

/* gets a set of values given a path */
export function getSet(items, path) {
  const results = new Set()
  if (Array.isArray(items) && typeof path === 'string') {
    items.forEach(item => {
      const value = get(item, path)
      if (value && value !== '') {
        if (Array.isArray(value)) {
          value.forEach(subitem => {
            if (isFalse(subitem) === false) {
              results.add(subitem)
            }
          })
        } else {
          if (isFalse(value) === false) {
            results.add(value)
          }
        }
      }
    })
  }
  return results
}

export function getLowerSet(items, path) {
  const results = new Set()
  if (Array.isArray(items) && typeof path === 'string') {
    items.forEach(item => {
      const value = get(item, path)
      if (value && value !== '') {
        if (Array.isArray(value)) {
          value.forEach(subitem => {
            if (isFalse(subitem) === false) {
              results.add(subitem.toLowerCase())
            }
          })
        } else {
          if(isFalse(value) === false) {
            results.add(value.toLowerCase())
          }
        }
      }
    })
  }
  return results
}

export function getFilterData(key, path, currentSearchResults, filters) {
  if (currentSearchResults && filters && filters[key]) {
    const names = getLowerSet(currentSearchResults.repos, path)
    return filters[key].filter(({ name, value }) => {
      return names.has(normalize(name)) || names.has(normalize(value))
    })
  }
}

export function hasLicense(repo) {
  return repo.permissions && Array.isArray(repo.permissions.licenses) && repo.permissions.licenses.length > 0
}

export function clean(string) {
  try {
    return string.toLowerCase().replace(/-/g,' ').trim()
  } catch (error) {
    return string
  }
}

export function sortByBestMatch(a, b) {
  return Number(a.searchScore) > Number(b.searchScore) ? -1 : 1
}

export function sortByName(a, b) {
  return clean(a.name) > clean(b.name) ? 1 : -1
}

export function getDate(repo, path) {

  if (repo.ghUpdatedAt) {
    return new Date(a.ghUpdatedAt).getTime()
  } else if (repo.date && repo.date.lastModified) {
    return new Date(a.date.lastModified).getTime()
  } else {
    return
  }
}

export function getTime(date) {
  try {
    if (date) {
      return new Date(date).getTime()
    } else {
      return -10e10
    }
  } catch (error) {
    return -10e10
  }
}

export function sortByDate(a, b) {
  try {
    const aDate = get(a, 'date.lastModified') || get(a, 'ghUpdatedAt')
    const bDate = get(b, 'date.lastModified') || get(b, 'ghUpdatedAt')
    const aTime = getTime(aDate)
    const bTime = getTime(bDate)
    return Math.sign(bTime - aTime) || 0;
  } catch (error) {
    return 0
  }
}

export function sortByDataQuality(a, b) {
  return Number(a.score) > Number(b.score) ? -1 : 1
}