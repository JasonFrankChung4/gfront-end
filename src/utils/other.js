/* global ASSET_PATH */
/* global URLSearchParams */

import get from 'lodash.get'
import intersection from 'lodash.intersection'
import { lower, run, startsWith, trim } from '@code.gov/cautious'
import siteConfig from '../../config/site/site.json'

export const falses = [undefined, null, 'null', 'None', 'Null', 'NULL', '', 'False', 'false']

export function isFalse(input) {
  return falses.includes(input)
}


export function adjustAssetPath(thing) {
  const pattern = /.?\/?assets\//
  const newAssetPath = ASSET_PATH + 'assets/'
  if (startsWith(thing, './assets') || startsWith(thing, '/assets/') || startsWith(thing, 'assets/')) {
    return thing.replace(pattern, newAssetPath)
  } else if (typeof value === 'object') {
    for (let key in thing) {
      const subvalue = thing[key]
      if (typeof subvalue === 'string') {
        thing[key] = adjustAssetPath(subvalue)
      }
    }
    return thing
  } else {
    return thing
  }
}

export function getConfigValue(path) {
  if (siteConfig) {
    let value = get(siteConfig, path)
    if (typeof value === 'string') {
      try {
        value = adjustAssetPath(value)
      } catch (error) {
        console.error("couldn't check if the following started with assets", [value])
      }
    } else if (Array.isArray(value)) {
      value = value.map(item => {
        if (typeof item === 'object') {
          for (let subkey in item) {
            let subvalue = item[subkey]
            if (typeof subvalue === 'string') {
              item[subkey] = adjustAssetPath(subvalue)
            }
          }
        }
        return item
      });
    }
    if (!value) {
      console.warn(`We weren't able to find the value for ${path} in your code-gov-config.json file.
      You can consult examples under the folder found at config/site/examples
      or consult our developer documentation here: https://developers.code.gov/configure.html`)
    }
    return value
  }
  else {
    return null
  }
}

export function normalize(input) {
  return trim(lower(input))
}

/* runs when each page component is loaded */
export function refreshView() {
  window.scrollTo(0, 0)
  document.activeElement.blur()
}


export function scrollToTopOfResults() {
  const headerHeight = document.querySelector("header.main").clientHeight
  const bannerHeight = document.querySelector(".banner").clientHeight
  const navHeight = document.querySelector("header.main nav").clientHeight
  const scrollDepth = headerHeight + bannerHeight - navHeight
  window.scrollTo(0, scrollDepth)
}

/* gets a set of values given a path */
export function getSet(items, path) {
  const results = new Set()
  if (Array.isArray(items) && typeof path === 'string') {
    items.forEach(item => {
      const value = get(item, path)
      if (value && value !== '') {
        if (Array.isArray(value)) {
          value.forEach(subitem => {
            if (isFalse(subitem) === false) {
              results.add(subitem)
            }
          })
        } else {
          if (isFalse(value) === false) {
            results.add(value)
          }
        }
      }
    })
  }
  return results
}

export function getLowerSet(items, path) {
  const results = new Set()
  if (Array.isArray(items) && typeof path === 'string') {
    items.forEach(item => {
      const value = get(item, path)
      if (value && value !== '') {
        if (Array.isArray(value)) {
          value.forEach(subitem => {
            if (isFalse(subitem) === false) {
              results.add(subitem.toLowerCase())
            }
          })
        } else {
          if(isFalse(value) === false) {
            results.add(value.toLowerCase())
          }
        }
      }
    })
  }
  return results
}

export function getFilterData(key, path, currentSearchResults, filters) {
  if (currentSearchResults && filters && filters[key]) {
    const names = getLowerSet(currentSearchResults.repos, path)
    return filters[key].filter(({ name, value }) => {
      return names.has(normalize(name)) || names.has(normalize(value))
    })
  }
}

export function hasLicense(repo) {
  return repo.permissions && Array.isArray(repo.permissions.licenses) && repo.permissions.licenses.length > 0
}


