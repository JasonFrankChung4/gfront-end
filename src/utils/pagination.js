import { last, range } from '@code.gov/cautious'

export function getPageInfo(params) {
  const { count, pagesize } = params
  const pagecount = Math.ceil(count / pagesize)
  const pageIndexes = range(pagecount)
    .map(n => n + 1) // convert from starting at 0 to 1
  const pageCount = pageIndexes.length

  return { pageIndexes, pageCount }
}

export function getPaginationNavInfo(params) {
  const { pageIndexes, page } = params
  const ultimate = last(pageIndexes)
  const left = page - 1
  const right = page + 1

  return { ultimate, left, right }
}
