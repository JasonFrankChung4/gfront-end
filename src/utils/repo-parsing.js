import get from 'lodash.get'
import { filter } from '@code.gov/cautious'

export function getDisplayTextForUsageType(repo) {
  let text = ''
  const usageType = get(repo, 'permissions.usageType')
  if (usageType === 'openSource') {
    text = 'Open Source'
  } else if (usageType === 'governmentWideReuse') {
    text = 'Gov-wide Reuse'
  } else if (usageType) {
    text = usageType
  } else {
    text = 'unknown'
  }
  return text
}

export function getLicenseName(repo) {
  return get(repo, 'permissions.licenses[0].name')
}

export function getLaborHours(repo) {
  try {
    const laborHours =  Number(get(repo, 'contact.laborHours'))
    if (laborHours > 0) {
      return laborHours
    }
  } catch (error){
    console.warn(error)
  }
}

export function parseLanguages(repo) {
  const languages = get(repo, 'languages')
  if (Array.isArray(languages)) {
    return filter(languages, Boolean)
  } else {
    return []
  }
}

export function parseEmail(repo) {
  const email = get(repo, 'contact.email')
  if (Boolean(email)) {
    return email
  }
}

export function parseRepositoryURL(repo) {
  const url = get(repo, 'repositoryURL')
  if (Boolean(url)) {
    return url
  }
}

export function parseTags(repo) {
  try {
    return get(repo, 'tags').filter(Boolean)
  } catch (error) {
    return []
  }
}


export function getLastModifiedDateString(repo) {
  try {
    const dateLastModified = get(repo, 'date.lastModified') ||  get(repo, 'ghUpdatedAt')
    if (dateLastModified) {
      return new Date(dateLastModified).toLocaleDateString()
    }
  } catch (error) {
    console.warn(error)
  }
}
