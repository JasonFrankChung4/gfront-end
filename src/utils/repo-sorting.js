export function clean(string) {
  try {
    return string.toLowerCase().replace(/-/g,' ').trim()
  } catch (error) {
    return string
  }
}

export function getDate(repo, path) {
  if (repo.ghUpdatedAt) {
    return new Date(repo.ghUpdatedAt).getTime()
  } else if (repo.date && repo.date.lastModified) {
    return new Date(repo.date.lastModified).getTime()
  } else {
    return
  }
}

export function getTime(date) {
  try {
    if (date) {
      return new Date(date).getTime()
    } else {
      return -10e10
    }
  } catch (error) {
    return -10e10
  }
}

export function sortByBestMatch(a, b) {
  return Number(a.searchScore) > Number(b.searchScore) ? -1 : 1
}

export function sortByDataQuality(a, b) {
  return Number(a.score) > Number(b.score) ? -1 : 1
}

export function sortByDate(a, b) {
  try {
    const aDate = get(a, 'date.lastModified') || get(a, 'ghUpdatedAt')
    const bDate = get(b, 'date.lastModified') || get(b, 'ghUpdatedAt')
    const aTime = getTime(aDate)
    const bTime = getTime(bDate)
    return Math.sign(bTime - aTime) || 0;
  } catch (error) {
    return 0
  }
}

export function sortByName(a, b) {
  return clean(a.name) > clean(b.name) ? 1 : -1
}
