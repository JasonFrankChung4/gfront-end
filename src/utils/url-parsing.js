/* global URLSearchParams */

export function getParamAsArray(params, key) {
  if (params.has(key)) {
    return params.get(key)
      .split(',')
      .map(item => item.toLowerCase().trim())
  } else {
    return null
  }
}

export function getParamAsNumber(params, key) {
  if (params.has(key)) {
    return Number(params.get(key).toLowerCase().trim())
  } else {
    return null
  }
}

export function getParamAsString(params, key) {
  if (params.has(key)) {
    return params.get(key).toLowerCase().trim()
  } else {
    return null
  }
}

export function parseLocation(inpt) {
  const loc = inpt || window.location
  const pathname = loc.pathname
  const params = new URLSearchParams(loc.search)

  return {
    pathname,
    agencies: getParamAsArray(params, 'agencies'),
    languages: getParamAsArray(params, 'languages'),
    licenses: getParamAsArray(params, 'licenses'),
    skillLevels: getParamAsArray(params, 'skillLevels'),
    timeRequired: getParamAsArray(params, 'timeRequired'),
    page: getParamAsNumber(params, 'page'),
    sort: getParamAsString(params, 'sort'),
    query: getParamAsString(params, 'query')
  }
}
