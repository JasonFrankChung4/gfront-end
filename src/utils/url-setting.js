export function setArrayAsParam(urlSearchParams, key, value) {
  if (urlSearchParams && Array.isArray(value) && value.length > 0) {
    urlSearchParams.set(key, value.join(','))
  }
}

export function setStringAsParam(urlSearchParams, key, value) {
  if (urlSearchParams && typeof value === 'string' && value.length > 0) {
    urlSearchParams.set(key, value)
  }
}

export function setNumberAsParam(urlSearchParams, key, value) {
  if (urlSearchParams && typeof value === 'number') {
    urlSearchParams.set(key, value)
  }
}