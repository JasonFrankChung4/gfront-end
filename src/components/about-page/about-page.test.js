import React from 'react'
import { shallow } from 'enzyme'

import client from 'api-client'
import AboutPage from 'components/about-page'
import ComplianceDashboard from 'components/about-page/compliance-dashboard.component'
import InventoryCode from 'components/about-page/inventory-code.component'
import JSONValidator from 'components/about-page/json-validator.component'
import { eventMap, push } from '../../../tests/mocks/window'
import { refreshView, scrollToTopOfResults, loadScript } from '../../utils/other'

jest.mock('utils/other')
jest.spyOn(window.customElements, 'get')

// helper for testing the component does not make changes if it is `loading`
const testLoading = component => {
  jest.spyOn(component, 'setState')
  component.loading = true
  component.componentDidMount()
  expect(component.setState).not.toBeCalled()
}

// helper for testing the webcomponent loads if it does not exist
const testWebcomponentLoads = component => {
  window.customElements.get.mockImplementation(() => undefined)
  loadScript.mockReset()
  component.loading = false
  component.componentDidMount()
  expect(loadScript).toBeCalled()
}

// helper for testing the webcomponent does not load if it already exists
const testWebcomponentExists = component => {
  window.customElements.get.mockImplementation(() => 'component-exists')
  loadScript.mockReset()
  component.loading = false
  component.componentDidMount()
  expect(loadScript).not.toBeCalled()
}

const compliances = [
  { name: 'agency-1-name', acronym: 'DOD', img: 'agency-1-img' },
  { name: 'agency-2-name', acronym: 'DOE', img: 'agency-2-img' },
  { name: 'agency-3-name', acronym: 'agency-3-acronym', img: 'agency-3-img' } // not on dashboard list
]

const props = {}

let wrapper
let instance
describe('components - AboutPage', () => {
  beforeEach(() => {
    wrapper = shallow(<AboutPage {...props} />)
    instance = wrapper.instance()
  })

  describe('componentDidMount', () => {
    it('should refresh the view', () => {
      expect(refreshView).toBeCalled()
    })

    it('should attach a `popstate` event listener for scrolling to the top of results', () => {
      // pathname does not start with `policy-guide/`, no scrolling
      eventMap.popstate()
      expect(scrollToTopOfResults).not.toBeCalled()
      // pathname starts with `PUBLIC_PATH + 'about'`, scroll to top
      push(`${PUBLIC_PATH}about/test-1`)
      eventMap.popstate()
      expect(scrollToTopOfResults).toBeCalled()
    })
  })

  describe('onNavChange', () => {
    it('should scroll to the top of the results', () => {
      instance.onNavChange()
      expect(scrollToTopOfResults).toBeCalled()
    })
  })

  describe('render', () => {
    it('should render correctly', () => {
      expect(wrapper).toMatchSnapshot()
    })
  })
})

describe('components - AboutPage - ComplianceDashboard', () => {
  beforeEach(() => {
    client.getCompliance.mockResolvedValue(compliances)
    wrapper = shallow(<ComplianceDashboard />)
    instance = wrapper.instance()
  })

  describe('componentDidMount', () => {
    it('should do nothing if it is `loading`', () => {
      testLoading(instance)
    })

    it('should load the webcomponent if it does not exist', () => {
      testWebcomponentLoads(instance)
    })

    it('should not try to load the webcomponent if exists', () => {
      testWebcomponentExists(instance)
    })

    it('should load the compliances into local state', () => {
      const expected = [
        expect.objectContaining({ acronym: 'DOD' }),
        expect.objectContaining({ acronym: 'DOE' })
      ]
      expect(wrapper.state('compliance')).toEqual(expected)
    })
  })

  describe('render', () => {
    it('should render correctly', () => {
      expect(wrapper).toMatchSnapshot()
    })
  })
})

describe('components - AboutPage - InventoryCode', () => {
  beforeEach(() => {
    wrapper = shallow(<InventoryCode />)
    instance = wrapper.instance()
    jest.spyOn(instance, 'setState')
  })

  describe('componentDidMount', () => {
    it('should do nothing if it is `loading`', () => {
      testLoading(instance)
    })

    it('should load the webcomponent if it does not exist', () => {
      testWebcomponentLoads(instance)
    })

    it('should not try to load the webcomponent if exists', () => {
      testWebcomponentExists(instance)
    })
  })

  describe('render', () => {
    it('should render correctly', () => {
      expect(wrapper).toMatchSnapshot()
    })
  })
})

describe('components - AboutPage - JSONValidator', () => {
  beforeEach(() => {
    wrapper = shallow(<JSONValidator />)
    instance = wrapper.instance()
    jest.spyOn(instance, 'setState')
  })

  describe('componentDidMount', () => {
    it('should do nothing if it is `loading`', () => {
      testLoading(instance)
    })

    it('should load the webcomponent if it does not exist', () => {
      testWebcomponentLoads(instance)
    })

    it('should not try to load the webcomponent if exists', () => {
      testWebcomponentExists(instance)
    })
  })

  describe('render', () => {
    it('should render correctly', () => {
      expect(wrapper).toMatchSnapshot()
    })
  })
})
