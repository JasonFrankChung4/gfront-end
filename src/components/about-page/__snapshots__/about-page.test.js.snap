// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components - AboutPage - ComplianceDashboard render should render correctly 1`] = `
<Fragment>
  <LazyHTML
    url="/src/components/about-page/html/compliance/agency-compliance.html"
  />
  <ComplianceDashboardComponent
    config={
      Object {
        "scores": Object {
          "compliant": Array [
            1,
            null,
          ],
          "noncompliant": Array [
            null,
            0.244444444,
          ],
          "partial": Array [
            0.25,
            0.9999999,
          ],
        },
        "text": Array [
          Object {
            "req": "agencyWidePolicy",
            "variants": Object {
              "compliant": "Agency policy is consistent with the Federal Source Code Policy.",
              "noncompliant": "Agency policy has not been reviewed for consistency with the Federal Source Code Policy.",
              "partial": "Agency policy is being updated for consistency with the Federal Source Code Policy.",
            },
          },
          Object {
            "req": "openSourceRequirement",
            "variants": Object {
              "compliant": "Agency has open sourced greater than 20% of their custom developed code.",
              "noncompliant": "Agency has open sourced less than 10% of their custom developed code.",
              "partial": "Agency has open sourced greater than 10% of their custom developed code.",
            },
          },
          Object {
            "req": "inventoryRequirement",
            "variants": Object {
              "compliant": "Agency has inventoried 100% of new custom code.",
              "noncompliant": "Agency has inventoried less than 50% of new custom code.",
              "partial": "Agency has inventoried more than 50% of new custom code.",
            },
          },
        ],
      }
    }
    data={
      Array [
        Object {
          "acronym": "DOD",
          "img": "/assets/img/logos/agencies/DOD-50x50.png",
          "name": "agency-1-name",
        },
        Object {
          "acronym": "DOE",
          "img": "/assets/img/logos/agencies/DOE-50x50.png",
          "name": "agency-2-name",
        },
      ]
    }
  />
</Fragment>
`;

exports[`components - AboutPage - InventoryCode render should render correctly 1`] = `
<Fragment>
  <LazyHTML
    url="/src/components/about-page/html/compliance/how-to-inventory-a.html"
  />
  <InventoryCodeComponent
    schema={Object {}}
    url="/assets/data/schema.json"
  />
  <LazyHTML
    url="/src/components/about-page/html/compliance/how-to-inventory-b.html"
  />
</Fragment>
`;

exports[`components - AboutPage render should render correctly 1`] = `
<div>
  <Connect(SiteBanner)
    title="ABOUT"
  />
  <Breadcrumbs
    crumbs={
      Array [
        Object {
          "text": "Home",
          "to": "/",
        },
        Object {
          "text": "About",
        },
      ]
    }
  />
  <br />
  <div
    className="indented"
  >
    <div
      className="show-w-lte-600"
      style={
        Object {
          "padding": "30px",
          "textAlign": "center",
        }
      }
    >
      <Connect(NavSelect)
        pages={
          Array [
            Object {
              "display": "Overview - Introduction",
              "route": "/about/overview/introduction",
            },
            Object {
              "display": "Overview - Tracking Progress",
              "route": "/about/overview/tracking-progress",
            },
            Object {
              "display": "Compliance - Agency Compliance",
              "route": "/about/compliance/dashboard",
            },
            Object {
              "display": "Compliance - How to Procure Software",
              "route": "/about/compliance/procurement",
            },
            Object {
              "display": "Compliance - How to Inventory",
              "route": "/about/compliance/inventory-code",
            },
            Object {
              "display": "Compliance - Validate Schema",
              "route": "/about/compliance/inventory-code/validate-schema",
            },
            Object {
              "display": "Open Source Pilot - Introduction",
              "route": "/about/open-source/introduction",
            },
            Object {
              "display": "Open Source Pilot - Tools and Resources",
              "route": "/about/open-source/resources",
            },
            Object {
              "display": "Open Source Pilot - How to Measure",
              "route": "/about/open-source/measuring-code",
            },
            Object {
              "display": "Open Source Pilot - Licensing",
              "route": "/about/open-source/licensing",
            },
          ]
        }
      />
    </div>
    <div
      className="width-quarter show-w-gt-600 sticky"
    >
      <SideNav
        alignment="left"
        baseurl="/about"
        links={
          Array [
            Object {
              "children": Array [
                Object {
                  "route": "/overview/introduction",
                  "text": "Introduction",
                },
                Object {
                  "route": "/overview/tracking-progress",
                  "text": "Tracking Progress",
                },
              ],
              "route": "/overview/introduction",
              "text": "Overview",
            },
            Object {
              "children": Array [
                Object {
                  "route": "/compliance/dashboard",
                  "text": "Agency Compliance",
                },
                Object {
                  "route": "/compliance/procurement",
                  "text": "How to Procure Software",
                },
                Object {
                  "route": "/compliance/inventory-code",
                  "text": "How to Inventory",
                },
              ],
              "route": "/compliance/dashboard",
              "text": "Compliance",
            },
            Object {
              "children": Array [
                Object {
                  "route": "/open-source/introduction",
                  "text": "Introduction",
                },
                Object {
                  "route": "/open-source/resources",
                  "text": "Tools and Resources",
                },
                Object {
                  "route": "/open-source/measuring-code",
                  "text": "How to Measure",
                },
                Object {
                  "route": "/open-source/licensing",
                  "text": "Licensing",
                },
              ],
              "route": "/open-source/introduction",
              "text": "Open Source Pilot",
            },
          ]
        }
        onLinkClick={[Function]}
      />
    </div>
    <div
      className="docs-content"
    >
      <Switch>
        <Route
          component={[Function]}
          path="/about/overview/introduction"
        />
        <Route
          component={[Function]}
          path="/about/overview/tracking-progress"
        />
        <Redirect
          from="/about/overview"
          push={false}
          to="/about/overview/introduction"
        />
        <Route
          component={[Function]}
          path="/about/compliance/dashboard"
        />
        <Route
          component={[Function]}
          path="/about/compliance/procurement"
        />
        <Route
          component={[Function]}
          path="/about/compliance/inventory-code"
        />
        <Redirect
          from="/about/compliance"
          push={false}
          to="/about/compliance/dashboard"
        />
        <Route
          component={[Function]}
          path="/about/open-source/introduction"
        />
        <Route
          component={[Function]}
          path="/about/open-source/resources"
        />
        <Route
          component={[Function]}
          path="/about/open-source/measuring-code"
        />
        <Route
          component={[Function]}
          path="/about/open-source/licensing"
        />
        <Redirect
          from="/about/open-source"
          push={false}
          to="/about/open-source/introduction"
        />
      </Switch>
    </div>
  </div>
  <br />
  <br />
</div>
`;

exports[`render should render correctly 1`] = `
<Fragment>
  <LazyHTML
    url="/src/components/about-page/html/compliance/how-to-inventory-a.html"
  />
  <InventoryCodeComponent
    schema={Object {}}
    url="/assets/data/schema.json"
  />
  <LazyHTML
    url="/src/components/about-page/html/compliance/how-to-inventory-b.html"
  />
</Fragment>
`;
